pkg/modelzooserver/modelzooserver.go

```diff
@@ -275,54 +276,92 @@ func (s *modelZooServer) DropModelDef(ctx context.Context, req *pb.ModelDefReque
 	return &pb.ModelResponse{Success: true, Message: ""}, nil
 }
 
-func (s *modelZooServer) ReleaseTrainedModel(ctx context.Context, req *pb.TrainedModelRequest) (*pb.ModelResponse, error) {
+func (s *modelZooServer) ReleaseTrainedModel(stream pb.ModelZooServer_ReleaseTrainedModelServer) error {
+	var req *pb.TrainedModelRequest
+	var err error
+	var sqlf io.WriteCloser
+
+	// sqlf is a sqlfs writer, it will be created when the first stream request arrives.
+	// the uploaded model contents into MySQL using package sqlfs.
+	sqlf = nil
+
+	// Create database sqlflow_public_models to store public trained models.
+	if _, err := s.DB.Exec("CREATE DATABASE IF NOT EXISTS sqlflow_public_models;"); err != nil {
+		return err
+	}
+
+	for { // read stream request
+		// NOTE: other fields in req must be the same in every stream request.
+		streamReq, err := stream.Recv()
```

Comment:
variable `streamReq` seems unnecessary

Reply:
Well, it's needed as a temporary variable. If `err == io.EOF`, the `streamReq` is actually `nil`, so we need to get the last  none nil object as `req`.